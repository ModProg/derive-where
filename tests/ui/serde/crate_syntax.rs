use std::marker::PhantomData;

use derive_where::derive_where;

#[derive_where(Deserialize)]
#[serde(crate)]
struct WrongCrateSyntax1<T>(PhantomData<T>);

#[derive_where(Deserialize)]
#[serde(crate())]
struct WrongCrateSyntax2<T>(PhantomData<T>);

#[derive_where(Deserialize)]
#[serde(crate = ())]
struct NotString<T>(PhantomData<T>);

#[derive_where(Deserialize)]
#[serde(crate = "1")]
struct NotPath<T>(PhantomData<T>);

#[derive_where(Deserialize)]
#[serde(crate = "serde_")]
#[serde(crate = "serde_")]
struct Duplicate<T>(PhantomData<T>);

fn main() {}
