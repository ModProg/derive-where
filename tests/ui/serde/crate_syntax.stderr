error: unexpected option syntax
 --> tests/ui/serde/crate_syntax.rs:6:9
  |
6 | #[serde(crate)]
  |         ^^^^^

error: unexpected option syntax
  --> tests/ui/serde/crate_syntax.rs:10:9
   |
10 | #[serde(crate())]
   |         ^^^^^^^

error: unexpected option syntax
  --> tests/ui/serde/crate_syntax.rs:14:17
   |
14 | #[serde(crate = ())]
   |                 ^^

error: expected path, expected identifier
  --> tests/ui/serde/crate_syntax.rs:18:17
   |
18 | #[serde(crate = "1")]
   |                 ^^^

error: duplicate `crate` option
  --> tests/ui/serde/crate_syntax.rs:23:9
   |
23 | #[serde(crate = "serde_")]
   |         ^^^^^^^^^^^^^^^^
